gamemodes/terrortown/gamemode/server/sv_player_ext.lua

```diff
@@ -1625,27 +1628,36 @@ end
 ---
 -- Caches the weapons currently in the player inventory and removes them.
 -- These weapons can be restored at any time.
+-- @param boolean forceAll force all weapons to be removed, including weapon_ttt_unarmed
 -- @note As long as a player has cached weapons, they are unable to pick up any weapon.
 -- @realm server
-function plymeta:CacheAndStripWeapons()
-	local cachedWeaponInventory = {}
+function plymeta:CacheAndStripWeapons(forceAll)
+	self.cachedWeaponInventory = {}
+	self.cachedWeaponSelected = WEPS.GetClass(self:GetActiveWeapon())
 
 	local weps = self:GetWeapons()
 
 	for i = 1, #weps do
 		local wep = weps[i]
+		local wepClass = WEPS.GetClass(wep)
 
-		cachedWeaponInventory[#cachedWeaponInventory + 1] = {
-			cls = WEPS.GetClass(wep),
+		if not forceAll and wepClass == "weapon_ttt_unarmed" then continue end
+
+		self.cachedWeaponInventory[#self.cachedWeaponInventory + 1] = {
+			cls = wepClass,
 			clip1 = wep:Clip1(),
 			clip2 = wep:Clip2()
 		}
 	end
 
-	self.cachedWeaponInventory = cachedWeaponInventory
-	self.cachedWeaponSelected = WEPS.GetClass(self:GetActiveWeapon())
 
+	-- we have to use this hack here instead of StripWeapon because StripWeapon calls
+	-- OnDrop which is not intended for the weapon caching
 	self:StripWeapons()
+
+	if not forceAll then
+		self:Give("weapon_ttt_unarmed")
+	end
```

Comment:
this feels somewhat weird, but i guess it's the only way to do this :/

Reply:
I had loads of issues otherwise. I agree with you, but I was unable to find a different solution

Reply:
the biggest issue i have with this is if something deliberately removes the unarmed weapon in another context. Because then this might lead to confusing interactions.

Can we guard against this? or do we just assume that nobody will remove the unarmed weapon?

Reply:
if someone removed the unarmed weapon I think they have to handle it themselves